micro-rdk/src/common/ota.rs

```diff
@@ -334,14 +327,34 @@ impl<S: OtaMetadataStorage> OtaService<S> {
             uri = hyper::Uri::from_parts(parts).map_err(|e| OtaError::Other(e.to_string()))?;
         };
 
+        Ok(uri)
+    }
+
+    /// Attempts to perform an OTA update.
+    /// On success, returns an `Ok(true)` or `Ok(false)` indicating if a reboot is necessary.
+    pub(crate) async fn update(&mut self) -> Result<bool, OtaError<S>> {
+        if !(self.needs_update()) {
+            return Ok(false);
+        }
+
+        let mut uri = self.parse_uri(&self.url)?;
+        let mut response: hyper::Response<hyper::body::Incoming>;
+        let mut conn;
+
         let mut num_tries = 0;
-        let (mut sender, conn) = loop {
+        loop {
             num_tries += 1;
             if num_tries == NUM_RETRY_CONN + 1 {
                 return Err(OtaError::Other(
                     "failed to establish connection".to_string(),
                 ));
             }
+
+            log::info!("OTA connection attempt {}: `{}` ", num_tries, uri);
+
+            let mut sender = None;
+            let mut inner_conn = None;
+            let retry;
```

Comment:
Last thing: I'm not convinced you need the `retry` flag at all. If `sender` and `inner_conn` are set, then you can proceed, otherwise it is an error. Why not just check for `if sender.is_none() || inner_conn.is_none` to decide to retry?